source/simulator/core.cc

```diff
@@ -2421,6 +2399,11 @@ namespace aspect
         // then do the core work: assemble systems and solve
         solve_timestep ();
 
+        const bool statistics_already_generated = (timestep_number == 0) && (pre_refinement_step == 0)
+            && (parameters.nonlinear_solver == NonlinearSolver::IMPES || parameters.nonlinear_solver == NonlinearSolver::Advection_only);
+        if (!statistics_already_generated)
+          generate_global_statistics();
```

Comment:
Okay, I made this easier to follow, as the `maybe_generate_statistics` function now returns what it did as a bool.